lib/portage/eapi.py

```diff
@@ -2,33 +2,54 @@
 # Distributed under the terms of the GNU General Public License v2
 
 import collections
-import operator
-import types
-
-from portage import eapi_is_supported
+from functools import lru_cache
 
 
+@lru_cache(None)
```

Comment:
I don't understand why we expect an lru_cache to help here. The function is trivial. Same goes for the rest of the eapi_has_* functions.

Reply:
I have not measured anything, but some quite trivial functions are called thousands or millions times by dependency resolver...

Reply:
The main intent of this commit is to replace [this custom cache implementation](https://github.com/gentoo/portage/blob/master/lib/portage/eapi.py#L322-L356) with the builtin one.

probably I should make this clearer in the commit message if it's still confusing.

Anyway, since the original implementation cached _every_ function in here, I did the same in order to provide a 1-to-1 mapping.

That being said, these types of conversations are _exactly_ the reasons why I wanted to do this sort of cleanup. By making it more clear what's going on, it makes it easier to question the utility of such things.